### PR TITLE
ENT-6091 | Update dependencies.

### PR DESCRIPTION
**Pull request by webbj74**
Although this package is not dependent on any other PHP packages, we currently have no way to determine if this package is compatible with a particular version of PHP. This PR introduces Travis-CI tests and some phpunit tests to allow for that. 

I used `SuppressWarning` annotations for PHP Mess Detector in a few places where safe refactoring suggests more test coverage (and this was supposed to be a quick update).
**Issue:** https://backlog.acquia.com/browse/ENT-6091
**Release Request:** https://backlog.acquia.com/browse/ENT-6161

### CODE REVIEW

**Comment by pbull** [comment]
@webbj74 we might want to look at codesniffer checks available in https://github.com/PHPCompatibility/PHPCompatibility to address the PHP version compatibility question. We have an existing story about this: https://backlog.acquia.com/browse/ENT-5515

File: src/AbstractLoader.php
Comment (pbull) [inline src/AbstractLoader.php]: `$config` is passed by reference; this doesn't actually have a return value.

File: src/AbstractLoader.php
Comment (pbull) [inline src/AbstractLoader.php]: It looks like this can be a relative or absolute path, can this description be expanded to note that.

**Review by codexmas — APPROVED** [review]
Code readability improvements are nice :)
Tests are passing yay!

**Review by aaronsenecal — APPROVED** [review]

**Review by pbull — APPROVED** [review]

**Comment by pbull** [comment]
@webbj74 looks like @aaronsenecal and/or I would need write access to this repo for our approvals to count.

**Pull request merged**
